colcon_in_container/verb/_rosdep.py

```diff
@@ -18,19 +18,31 @@
 from colcon_in_container.logging import logger
 
 
-def call_rosdep(provider, ros_distro, dependency_types: Set[str]):
-    """Call rosdep init, update and install provided dependency_types."""
-    # initialize and call rosdep over our repository
-    logger.info('Initialising and calling rosdep')
-    commands = [
-        'rosdep init',
-        'rosdep update',
-        # Avoid rosdep/apt interactive shell error message
-        'export DEBIAN_FRONTEND=noninteractive',
-        'rosdep install --from-paths /ws/src --ignore-src -y '
-        f'--rosdistro={ros_distro} '
-    ]
-    for dependency_type in dependency_types:
-        commands[-1] += f'--dependency-types={dependency_type} '
-
-    return provider.execute_commands(commands)
+class Rosdep(object):
+    """Rosdep tool class wrapper to call rosdep in a provider."""
+
+    def __init__(self, provider, ros_distro):
```

Comment:
What if you pass it a provider that's already rosdep init?

Reply:
For now nothing, since we are not managing the command return value